python/mol.cpp

```diff
@@ -176,9 +176,22 @@ void add_mol(py::module& m) {
                      (self.given ? " (" : " (not "), "given)>");
     });
 
+  py::class_<Assembly::Oper>(m, "AssemblyOper")
+    .def(py::init<>())
+    .def_readonly("name", &Assembly::Oper::name)
+    .def_readonly("type", &Assembly::Oper::type)
+    .def_readonly("transform", &Assembly::Oper::transform);
+
+  py::class_<Assembly::Gen>(m, "AssemblyGen")
+    .def(py::init<>())
+    .def_readonly("chains", &Assembly::Gen::chains)
+    .def_readonly("subchains", &Assembly::Gen::subchains)
+    .def_readonly("opers", &Assembly::Gen::opers);
+
   py::class_<Assembly>(m, "Assembly")
     .def_readonly("name", &Assembly::name)
     .def_readonly("oligomeric_details", &Assembly::oligomeric_details)
+    .def_readonly("generators", &Assembly::generators)
```

Comment:
Same applies to `std::vector<Assembly::Gen>` bindings.

Reply:
I don't really have opinion on this. I'll merge it now and I'll change it to Assembly.Gen, etc.
I think I'll also rename Oper to Operator and opers to operators (in both C++ and Python).

Reply:
My point here is that such bindings provide a way to modify a temporary without knowing that. It's generally not good if you can't tell what python code does unless you look at C++ sources).

Maybe this this particular place has a little chance of being modified by user, but in general such API oddities should be avoided. 

*(Sorry, I feel like I'm developing discussion out of nothing)*

Reply:
ok, I changed also that